pkgs/servers/matrix-appservice-discord/default.nix

```diff
@@ -1,35 +1,57 @@
-{ lib, mkYarnPackage, fetchFromGitHub, runCommand, makeWrapper, python3, nodejs }:
-
-assert lib.versionAtLeast nodejs.version "12.0.0";
+{ lib
+, mkYarnPackage
+, fetchYarnDeps
+, fetchFromGitHub
+, runCommand
+, makeWrapper
+, python3
+, nodejs
+, removeReferencesTo
+}:
 
 let
+  pin = lib.importJSON ./pin.json;
   nodeSources = runCommand "node-sources" {} ''
     tar --no-same-owner --no-same-permissions -xf ${nodejs.src}
     mv node-* $out
   '';
 
 in mkYarnPackage rec {
   pname = "matrix-appservice-discord";
-
-  # when updating, run `./generate.sh <git release tag>`
-  version = "1.0.0";
+  inherit (pin) version;
 
   src = fetchFromGitHub {
-    owner = "Half-Shot";
+    owner = "matrix-org";
     repo = "matrix-appservice-discord";
     rev = "v${version}";
-    sha256 = "0pca4jxxl4b8irvb1bacsrzjg8m7frq9dnx1knnd2n6ia3f3x545";
+    sha256 = pin.srcSha256;
   };
 
   packageJSON = ./package.json;
-  yarnNix = ./yarn-dependencies.nix;
+  offlineCache = fetchYarnDeps {
+    yarnLock = "${src}/yarn.lock";
+    sha256 = pin.yarnSha256;
+  };
 
   pkgConfig = {
+    "@matrix-org/matrix-sdk-crypto-nodejs" = {
+      # Building this native Rust dependency requires the `napi` tool from
+      # `napi-rs`, which isn't available in nixpkgs at the time of writing.
+      # Because appservice-discord does not really make use of this library
+      # anyway, we can snip it off to neutralise loading errors.
```

Comment:
This isn't true any more -- do you think it's worth copying the library, or no (since it's not used)?

Reply:
You unresolved this without responding -- mind sharing your opinion?

Reply:
My email reply landed outside of the thread it seems:
https://github.com/NixOS/nixpkgs/pull/186316#issuecomment-1282450133